xamarin/Xamarin.Forms
# Cross-Platform OS Theme APIs

### Description of Change ###

Adds the cross-platform APIs to support different OS Themes

#### Background
See #9804 for the relevant APIs and how to use it. Because of timing I have removed the dependency on Essentials from this (see #9926) so we can get this merged in time for 4.6. After that, we can easily swap out this implementation for the Essentials one. By keeping naming and enum values, etc. the same this should be an easy task.

### Issues Resolved ### 

- related to #9804

### API Changes ###

Added:
 - public enum AppTheme { Unspecified, Light, Dark }
 - public class AppThemeColor
 - public class OnAppTheme<T>
 - AppTheme Application.RequestedTheme { get; }
 - public event EventHandler<AppThemeChangedEventArgs> Application.RequestedThemeChanged;
 - public void Application.OnRequestedThemeChanged(AppThemeChangedEventArgs args)
 - public class AppThemeChangedEventArgs
 - protected virtual void OnRequestedThemeChanged(AppTheme newValue)

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android
- UWP
- MacOS
- WPF
- Tizen

### Behavioral/Visual Changes ###


None

### Before/After Screenshots ### 


Not applicable

### Testing Procedure ###

I have added a gallery page which shows (nearly) all ways to use this new functionality. Check if that works as expected. Whenever the values are used as a `DynamicResource` they should update whenever the theme is changed while the app is running. Check out the gallery page or, to take it a step further, download the NuGets and take it for a spin in your own (sample) app.

### PR Checklist ###
